### PR TITLE
add queryweb3

### PR DESCRIPTION
**Pull request by lechangxu**
#### Project Abstract

QueryWeb3 aims to address the fragmentation of information and data across various projects within the Polkadot and Kusama ecosystem. By aggregating data, we improve information retrieval and transaction efficiency, enhance user experience, and increase user satisfaction.

#### Grant [level](https://github.com/w3f/Grants-Program#level_slider-levels)

- [ ] **Level 1**:  Up to $10,000, 2 approvals
- [x] **Level 2**:  Up to $30,000, 3 approvals
- [ ] **Level 3**:  Unlimited, 5 approvals (for >$100k: Web3 Foundation Council approval)

#### Application Checklist

- [x] The [application template](https://github.com/w3f/Grants-Program/blob/master/applications/application-template.md) has been copied and aptly renamed (`project_name.md`).
- [x] I have read the [application guidelines](https://github.com/w3f/Grants-Program/blob/master/docs/Support%20Docs/grant_guidelines_per_category.md).
- [x] Payment details have been provided (Polkadot AssetHub (USDC & DOT) address in the application and bank details via email, if applicable).
- [x] I understand that an agreed upon percentage of each milestone will be paid in vested DOT, to the Polkadot address listed in the application.
- [x] I am aware that, in order to receive a grant, I (and the entity I represent) have to successfully complete a KYC/KYB check.
- [x] The software delivered for this grant will be released under an open-source license specified in the application.
- [x] The initial PR contains only one commit (squash and force-push if needed).
- [x] The grant will only be announced once the first milestone [has been accepted](https://github.com/w3f/Grant-Milestone-Delivery#process) (see the [announcement guidelines](https://github.com/w3f/Grants-Program/blob/master/docs/Support%20Docs/announcement-guidelines.md)).
- [x] I prefer the discussion of this application to take place in a private Element/Matrix channel. My username is: `@lechangxu :matrix.org` (change the homeserver if you use a different one)

### CODE REVIEW

**Comment by semuelle** [comment]
Thanks for the application, @lechangxu. I will review it as soon as possible (we are rather busy at the moment) and leave questions and/or notes in the chat room linked above.

**Comment by semuelle** [comment]
recheck

**Comment by lechangxu** [comment]
It‘s a great honor to hear from you. Waiting for your information.Thank you very much. Best wishes. Roger Le From: Sebastian Müller ***@***.***>Date: Tuesday, November 26, 2024 at 20:52To: w3f/Grants-Program ***@***.***>Cc: lechangxu ***@***.***>, Mention ***@***.***>Subject: Re: [w3f/Grants-Program] add queryweb3 (PR #2458)recheck—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you were mentioned.Message ID: ***@***.***>

File: applications/queryWeb3.md [hunk 1]
```diff
@@ -0,0 +1,287 @@
...
+
+This section should break the development roadmap down into milestones and deliverables. To assist you in defining it, we have created a document with examples for some grant categories [here](../docs/Support%20Docs/grant_guidelines_per_category.md). Since these will be part of the agreement, it helps to describe *the functionality we should expect in as much detail as possible*, plus how we can verify and test that functionality. Whenever milestones are delivered, we refer to this document to ensure that everything has been delivered as expected.
+
+Below we provide an **example roadmap**. In the descriptions, it should be clear how your project is related to Substrate, Kusama or Polkadot. We *recommend* that teams structure their roadmap as 1 milestone ≈ 1 month.
+
+> :exclamation: If any of your deliverables are based on somebody else's work, make sure you work and publish *under the terms of the license* of the respective project and that you **highlight this fact in your milestone documentation** and in the source code if applicable! **Projects that submit other people's work without proper attribution will be immediately terminated.**
```
Comment (semuelle) [inline applications/queryWeb3.md:181]:
```suggestion
```

File: applications/queryWeb3.md [hunk 2]
```diff
@@ -0,0 +1,287 @@
...
+
+- https://github.com/Query-Web3/backend
+- https://github.com/Query-Web3/frontend
+- https://github.com/Query-Web3/database
+
+### Team LinkedIn Profiles (if available)
+
+- https://www.linkedin.com/{person_1}
+- https://www.linkedin.com/{person_2}
+
```
Comment (semuelle) [inline applications/queryWeb3.md:169]:
```suggestion
```

File: applications/queryWeb3.md [hunk 3]
```diff
@@ -0,0 +1,287 @@
...
+- **Registered Address:** 3118 Yindu Road, Minhang District, Shanghai, China
+- **Registered Legal Entity:** N/A
+
+### Team's experience
+
+Please describe the team's relevant experience. If your project involves development work, we would appreciate it if you singled out a few interesting projects or contributions made by team members in the past.
+
+If anyone on your team has applied for a grant at the Web3 Foundation previously, please list the name of the project and legal entity here.
+
```
Comment (semuelle) [inline applications/queryWeb3.md:154]:
```suggestion
```

**Review by semuelle — CHANGES_REQUESTED** [review]

**Comment by lechangxu** [comment]
Already updated, please check. Thank you very much. Best regards. From: Sebastian Müller ***@***.***>Date: Friday, November 29, 2024 at 22:43To: w3f/Grants-Program ***@***.***>Cc: lechangxu ***@***.***>, Mention ***@***.***>Subject: Re: [w3f/Grants-Program] add queryweb3 (PR ***@***.*** requested changes on this pull request.In applications/queryWeb3.md:> +This section should break the development roadmap down into milestones and deliverables. To assist you in defining it, we have created a document with examples for some grant categories [here](../docs/Support%20Docs/grant_guidelines_per_category.md). Since these will be part of the agreement, it helps to describe *the functionality we should expect in as much detail as possible*, plus how we can verify and test that functionality. Whenever milestones are delivered, we refer to this document to ensure that everything has been delivered as expected.++Below we provide an **example roadmap**. In the descriptions, it should be clear how your project is related to Substrate, Kusama or Polkadot. We *recommend* that teams structure their roadmap as 1 milestone ≈ 1 month.++> :exclamation: If any of your deliverables are based on somebody else's work, make sure you work and publish *under the terms of the license* of the respective project and that you **highlight this fact in your milestone documentation** and in the source code if applicable! **Projects that submit other people's work without proper attribution will be immediately terminated.**⬇️ Suggested change -This section should break the development roadmap down into milestones and deliverables. To assist you in defining it, we have created a document with examples for some grant categories [here](../docs/Support%20Docs/grant_guidelines_per_category.md). Since these will be part of the agreement, it helps to describe *the functionality we should expect in as much detail as possible*, plus how we can verify and test that functionality. Whenever milestones are delivered, we refer to this document to ensure that everything has been delivered as expected.--Below we provide an **example roadmap**. In the descriptions, it should be clear how your project is related to Substrate, Kusama or Polkadot. We *recommend* that teams structure their roadmap as 1 milestone ≈ 1 month.--> :exclamation: If any of your deliverables are based on somebody else's work, make sure you work and publish *under the terms of the license* of the respective project and that you **highlight this fact in your milestone documentation** and in the source code if applicable! **Projects that submit other people's work without proper attribution will be immediately terminated.**In applications/queryWeb3.md:> +### Team LinkedIn Profiles (if available)++- https://www.linkedin.com/{person_1}+- https://www.linkedin.com/{person_2}+⬇️ Suggested change -### Team LinkedIn Profiles (if available)--- https://www.linkedin.com/{person_1}-- https://www.linkedin.com/{person_2}-In applications/queryWeb3.md:> +Please describe the team's relevant experience. If your project involves development work, we would appreciate it if you singled out a few interesting projects or contributions made by team members in the past.++If anyone on your team has applied for a grant at the Web3 Foundation previously, please list the name of the project and legal entity here.+⬇️ Suggested change -Please describe the team's relevant experience. If your project involves development work, we would appreciate it if you singled out a few interesting projects or contributions made by team members in the past.--If anyone on your team has applied for a grant at the Web3 Foundation previously, please list the name of the project and legal entity here-—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you were mentioned.Message ID: ***@***.***>

File: applications/queryWeb3.md [hunk 4]
```diff
@@ -0,0 +1,292 @@
...
+- data process: Initially, extract data from APIs like Subscan, Hydration, Bifrost and so on for DOT, KSM, Hydration, and Bifrost, process it into multidimensional BI data, and save it to MySQL to form the data source.
+
+- Business folw ： the same as use case2
```
Comment (takahser) [inline applications/queryWeb3.md:49]:
```suggestion
- Business flow: the same as use case 2
```

File: applications/queryWeb3.md [hunk 5]
```diff
@@ -0,0 +1,292 @@
...
+- Data process： Extract DEFI yield data through the APIs of various blockchains into MySQL and other databases, then process and organize it into multidimensional data suitable for BI, and store it in MySQL.
+
+- Business flow
+![Business flow for yield query](https://github.com/Query-Web3/product/blob/main/business%20flow/YieldQuery_businessflow.webp)
+
+- User Roles: Polkadot users including investors, operators, data analysts, technical personnel, and media personnel.
+
+![Yield query](https://github.com/Query-Web3/product/blob/main/UI/Yield%20query.webp)
+
+Use Case 3:Vol & Txns Query
```
Comment (takahser) [inline applications/queryWeb3.md:43]:
```suggestion
Use Case 3: Vol & Txns Query
```

File: applications/queryWeb3.md [hunk 6]
```diff
@@ -0,0 +1,292 @@
...
+
+![queryweb3](https://github.com/Query-Web3/product/blob/main/UI/queryweb3_with_AI.png)
+
+
+Use Case 2: Yield Query
+
+- Business Description：allows users to query snapshots of various chains within the Polkadot ecosystem (e.g., DOT, KSM, Hydration, Bifrost, etc.) on any given day. Users can search for all asset types (RWA or DeFi) and all yield types (staking, farming), including all tokens (DOT, KSM, HDT, vDOT, vKSM, BNC, etc.). The query results will list the TVL (Total Value Locked), 24-hour trading volume, and the number of transactions for each token or currency pair.  Additionally, based on these query results, our platform can display the top 10 yield rates in the Polkadot ecosystem, the top 10 chains by TVL, and various visual representations such as line charts, pie charts, or other graphical formats. This information is designed to provide investors and operators with valuable insights for making informed investment and operational decisions.
+
+- Data process： Extract DEFI yield data through the APIs of various blockchains into MySQL and other databases, then process and organize it into multidimensional data suitable for BI, and store it in MySQL.
```
Comment (takahser) [inline applications/queryWeb3.md:34]:
```suggestion
- Data process: Extract DEFI yield data through the APIs of various blockchains into MySQL and other databases, then process and organize it into multidimensional data suitable for BI, and store it in MySQL.
```

File: applications/queryWeb3.md [hunk 7]
```diff
@@ -0,0 +1,292 @@
...
+- **User Roles**: Polkadot users including investors, operators, data analysts, technical personnel, and media personnel.
+
+![Vol&Txns query](https://github.com/Query-Web3/product/blob/main/UI/vol%26txns%20query.webp)
+
+
+### Ecosystem Fit
+
+Help us locate your project in the Polkadot/Substrate/Kusama landscape and what problems it tries to solve by answering each of these questions:
+
+- Where and how does your project fit into the ecosystem?
```
Comment (takahser) [inline applications/queryWeb3.md:60]: Feel free to add a short answer here as well.

File: applications/queryWeb3.md [hunk 8]
```diff
@@ -0,0 +1,292 @@
...
+
+- Who is your target audience (parachain/dapp/wallet/UI developers, designers, your own user base, some dapp's userbase, yourself)?
+Our target audience includes parachain/dapp/wallet/UI developers, designers, our own user base, and users of other dapps within the ecosystem.
+
+- What need(s) does your project meet?
+QueryWeb3 meets the need for timely and integrated information from the Polkadot ecosystem, addressing issues such as finding high-yield financial products, getting timely feedback on information queries, tracking major wallet movements, and identifying airdrops and NFT rewards.
+
+- How did you identify these needs? Please provide evidence in the form of (scientific) articles, forum discussions, case studies, or raw data.
```
Comment (takahser) [inline applications/queryWeb3.md:68]: Feel free to add a short answer here as well.

**Review by takahser — CHANGES_REQUESTED** [review]
@lechangxu good to see you again here, after e-meeting you on Dr Cao's X space the other day. 👍

I have one initial request: Could you please reply directly in the GitHub thread, rather than replying via email? Currently, your comments are very verbose, and they're difficult to understand for that reason.

In addition to that, could you provide additional clarification on a few points related to your data sources and API integration?

In particular:

**1) Specific APIs and Endpoints**

- While you mention Subscan, Hydration, and Bifrost APIs, could you provide a more comprehensive list of APIs and endpoints you plan to integrate? Ideally, this should be directly included as part of the milestone deliverables.
- Are there any specific considerations for rate limits, scalability, or handling data discrepancies for these APIs?

**2) Integration Details**

- How do you plan to technically integrate with these APIs? For example, will you use custom parsers, standard SDKs, or third-party middleware?
- Could you elaborate on the methods you’ll use to validate and normalize the data collected from these sources? Or is that something that is not needed to train your model?

**3) Handling Other Data Categories**

- In Use Case 1, you mention aggregating data such as document data, technical code, TG (Telegram) data, DC (Discord) data, X (Twitter) data, media data, and video data. Could you clarify the sources and methods for collecting this information?

**Comment by lechangxu** [comment]
> @lechangxu good to see you again here, after e-meeting you on Dr Cao's X space the other day. 👍
> 
> I have one initial request: Could you please reply directly in the GitHub thread, rather than replying via email? Currently, your comments are very verbose, and they're difficult to understand for that reason.
> 
> In addition to that, could you provide additional clarification on a few points related to your data sources and API integration?
> 
> In particular:
> 
> **1) Specific APIs and Endpoints**
> 
> * While you mention Subscan, Hydration, and Bifrost APIs, could you provide a more comprehensive list of APIs and endpoints you plan to integrate? Ideally, this should be directly included as part of the milestone deliverables.
> * Are there any specific considerations for rate limits, scalability, or handling data discrepancies for these APIs?
> 
> **2) Integration Details**
> 
> * How do you plan to technically integrate with these APIs? For example, will you use custom parsers, standard SDKs, or third-party middleware?
> * Could you elaborate on the methods you’ll use to validate and normalize the data collected from these sources? Or is that something that is not needed to train your model?
> 
> **3) Handling Other Data Categories**
> 
> * In Use Case 1, you mention aggregating data such as document data, technical code, TG (Telegram) data, DC (Discord) data, X (Twitter) data, media data, and video data. Could you clarify the sources and methods for collecting this information?

**Comment by lechangxu** [comment]
> > @lechangxu good to see you again here, after e-meeting you on Dr Cao's X space the other day. 👍
> > I have one initial request: Could you please reply directly in the GitHub thread, rather than replying via email? Currently, your comments are very verbose, and they're difficult to understand for that reason.
> > In addition to that, could you provide additional clarification on a few points related to your data sources and API integration?
> > In particular:
> > **1) Specific APIs and Endpoints**
> > 
> > * While you mention Subscan, Hydration, and Bifrost APIs, could you provide a more comprehensive list of APIs and endpoints you plan to integrate? Ideally, this should be directly included as part of the milestone deliverables.
> > * Are there any specific considerations for rate limits, scalability, or handling data discrepancies for these APIs?
> > 
> > **2) Integration Details**
> > 
> > * How do you plan to technically integrate with these APIs? For example, will you use custom parsers, standard SDKs, or third-party middleware?
> > * Could you elaborate on the methods you’ll use to validate and normalize the data collected from these sources? Or is that something that is not needed to train your model?
> > 
> > **3) Handling Other Data Categories**
> > 
> > * In Use Case 1, you mention aggregating data such as document data, technical code, TG (Telegram) data, DC (Discord) data, X (Twitter) data, media data, and video data. Could you clarify the sources and methods for collecting this information?



> @lechangxu good to see you again here, after e-meeting you on Dr Cao's X space the other day. 👍
> 
> I have one initial request: Could you please reply directly in the GitHub thread, rather than replying via email? Currently, your comments are very verbose, and they're difficult to understand for that reason.
> 
> In addition to that, could you provide additional clarification on a few points related to your data sources and API integration?
> 
> In particular:
> 
> **1) Specific APIs and Endpoints**
> 
> * While you mention Subscan, Hydration, and Bifrost APIs, could you provide a more comprehensive list of APIs and endpoints you plan to integrate? Ideally, this should be directly included as part of the milestone deliverables.
> * Are there any specific considerations for rate limits, scalability, or handling data discrepancies for these APIs?
> 
> **2) Integration Details**
> 
> * How do you plan to technically integrate with these APIs? For example, will you use custom parsers, standard SDKs, or third-party middleware?
> * Could you elaborate on the methods you’ll use to validate and normalize the data collected from these sources? Or is that something that is not needed to train your model?
> 
> **3) Handling Other Data Categories**
> 
> * In Use Case 1, you mention aggregating data such as document data, technical code, TG (Telegram) data, DC (Discord) data, X (Twitter) data, media data, and video data. Could you clarify the sources and methods for collecting this information?

File: applications/queryWeb3.md (hunk 1 above)
Comment (lechangxu) [reply applications/queryWeb3.md]: Done

File: applications/queryWeb3.md (hunk 2 above)
Comment (lechangxu) [reply applications/queryWeb3.md]: Done

File: applications/queryWeb3.md (hunk 3 above)
Comment (lechangxu) [reply applications/queryWeb3.md]: Done

**Review by takahser — CHANGES_REQUESTED** [review]
@lechangxu thanks for your reply.

> Due to the workload considerations and the $30,000 budget limit, we will initially support data extraction from these specific chains to complete use case 2. 

In general I like the way you're choosing a lean approach by integrating a limited number of APIs initially.
However, would it be possible to specify that accordingly in the proposal? Please add a bit more details, on which APIs specific to these chains you're planning to support.

> In subsequent proposals, we will improve and extract data from other chains. If possible, could we increase the budget by several tens of thousands of dollars in this proposal? This would allow us to include data extraction from other chains in the Polkadot ecosystem and also complete use case

It's up to you to request the amount you deem appropriate. But usually the committee is not keen of supporting level 3 grants for applicants that we haven't supported in the past. That said, you haven't answered the questions I've asked [in my previous comment](https://github.com/w3f/Grants-Program/pull/2458#pullrequestreview-2472979830) in 2) and 3).

**Comment by lechangxu** [comment]
Please wait a few days, our team is currently in discussions. We will respond after we have communicated the relevant issues. Thank you.

**Comment by lechangxu** [comment]
> @lechangxu good to see you again here, after e-meeting you on Dr Cao's X space the other day. 👍
> 
> I have one initial request: Could you please reply directly in the GitHub thread, rather than replying via email? Currently, your comments are very verbose, and they're difficult to understand for that reason.
> 
> In addition to that, could you provide additional clarification on a few points related to your data sources and API integration?
> 
> In particular:
> 
> **1) Specific APIs and Endpoints**
> 
> * While you mention Subscan, Hydration, and Bifrost APIs, could you provide a more comprehensive list of APIs and endpoints you plan to integrate? Ideally, this should be directly included as part of the milestone deliverables.
> * Are there any specific considerations for rate limits, scalability, or handling data discrepancies for these APIs?
> 
> **2) Integration Details**
> 
> * How do you plan to technically integrate with these APIs? For example, will you use custom parsers, standard SDKs, or third-party middleware?
> * Could you elaborate on the methods you’ll use to validate and normalize the data collected from these sources? Or is that something that is not needed to train your model?
> 
> **3) Handling Other Data Categories**
> 
> * In Use Case 1, you mention aggregating data such as document data, technical code, TG (Telegram) data, DC (Discord) data, X (Twitter) data, media data, and video data. Could you clarify the sources and methods for collecting this information?

**3) Handling Other Data Categories**
Yes, I hope this product focuses on data from the Polkadot and KSM ecosystems, regardless of whether this data comes from X, Telegram, or DC channels. This will allow users on this platform not only to get timely investment information, discover opportunities within the Polkadot ecosystem, and help ecosystem project operators identify issues but also help users learn about Polkadot technology and understand its development trends, such as event information and financing information. This is what differentiates us from other AI searches. Although the data volume is large, in the era of big data, this is not a problem. However, it still depends on strong technical and financial support in the future.

**Comment by lechangxu** [comment]
> > > @lechangxu good to see you again here, after e-meeting you on Dr Cao's X space the other day. 👍
> > > I have one initial request: Could you please reply directly in the GitHub thread, rather than replying via email? Currently, your comments are very verbose, and they're difficult to understand for that reason.
> > > In addition to that, could you provide additional clarification on a few points related to your data sources and API integration?
> > > In particular:
> > > **1) Specific APIs and Endpoints**
> > > 
> > > * While you mention Subscan, Hydration, and Bifrost APIs, could you provide a more comprehensive list of APIs and endpoints you plan to integrate? Ideally, this should be directly included as part of the milestone deliverables.
> > > * Are there any specific considerations for rate limits, scalability, or handling data discrepancies for these APIs?
> > > 
> > > **2) Integration Details**
> > > 
> > > * How do you plan to technically integrate with these APIs? For example, will you use custom parsers, standard SDKs, or third-party middleware?
> > > * Could you elaborate on the methods you’ll use to validate and normalize the data collected from these sources? Or is that something that is not needed to train your model?
> > > 
> > > **3) Handling Other Data Categories**
> > > 
> > > * In Use Case 1, you mention aggregating data such as document data, technical code, TG (Telegram) data, DC (Discord) data, X (Twitter) data, media data, and video data. Could you clarify the sources and methods for collecting this information?
> 
> Answersing **1) Specific APIs and Endpoints**:
> 
> Due to the workload considerations and the $30,000 budget limit, we will initially support data extraction from these specific chains to complete use case 2. In subsequent proposals, we will improve and extract data from other chains. If possible, could we increase the budget by several tens of thousands of dollars in this proposal? This would allow us to include data extraction from other chains in the Polkadot ecosystem and also complete use case 3. I believe this approach would be better. So, could the W3F team please consider supporting my suggestion?
By the way, to avoid any misunderstanding, use case 1 and use case 3 are not included in this milestone. We will implement them in subsequent stages

File: applications/queryWeb3.md (hunk 4 above)
Comment (lechangxu) [reply applications/queryWeb3.md]: Done

File: applications/queryWeb3.md (hunk 5 above)
Comment (lechangxu) [reply applications/queryWeb3.md]: Done

File: applications/queryWeb3.md (hunk 6 above)
Comment (lechangxu) [reply applications/queryWeb3.md]: Done

File: applications/queryWeb3.md [hunk 9]
```diff
@@ -0,0 +1,304 @@
...
+- Business flow
+![business flow for queryweb3](https://github.com/Query-Web3/product/blob/main/business%20flow/queryweb3_with_ai_businessflow.png)
+- **User Roles**: Polkadot users
+
+![queryweb3](https://github.com/Query-Web3/product/blob/main/UI/queryweb3_with_AI.png)
+
+
+Use Case 2: Yield Query
+
+- Business Description：allows users to query snapshots of various chains within the Polkadot ecosystem (e.g., DOT, KSM, Hydration, Bifrost, etc.) on any given day. Users can search for all asset types (RWA or DeFi) and all yield types (staking, farming), including all tokens (DOT, KSM, HDT, vDOT, vKSM, BNC, etc.). The query results will list the TVL (Total Value Locked), 24-hour trading volume, and the number of transactions for each token or currency pair.  Additionally, based on these query results, our platform can display the top 10 yield rates in the Polkadot ecosystem, the top 10 chains by TVL, and various visual representations such as line charts, pie charts, or other graphical formats. This information is designed to provide investors and operators with valuable insights for making informed investment and operational decisions.
```
[the remark this answers is not in the document]
Comment (lechangxu) [reply applications/queryWeb3.md]: 1、Does that include dApps developed on parachains, e.g. Moonbeam dApps?
Yes, for Moonbeam, this phase will primarily implement StellaSwap, which currently has a TVL (Total Value Locked) of $16 million, accounting for the majority of the share.

2、Which dApps or protocols are going to be supported? Could you provide a list here?
Yes, this phase will primarily implement: Hydration, Bifrost, Acala, and Moonbeam. The data scope is as follows:
image

File: applications/queryWeb3.md (hunk 8 above)
Comment (lechangxu) [reply applications/queryWeb3.md:68]: I have been closely following and studying the Polkadot ecosystem for the past few years and have been an early investor in Polkadot. Based on my experience, I have encountered some pain points within the Polkadot ecosystem, which have prompted me to create a product to address these issues. Here are the details:

**Pain points**

1、How to find the highest yielding financial products at present? Which chains are they on?
2、If we raise an information question in the Polkadot ecosystem, can we get timely and effective feedback?
3、What are the major movements in the Polkadot ecosystem wallets? What have they done?
4、What activities are there in each of the Polkadot and kusama ecosystem projects?
5、Can we still send some NFT benefits to target users?
6、What airdrops are there in the Polkadot ecosystem projects? Which airdrops can I participate in?
7、Based on the Polkadot and Kusama ecosystem, which projects are in the top 10 for on-chain transaction volume and transaction amount on a weekly, monthly, quarterly, semi-annual, and annual basis? Additionally, what are their year-over-year and month-over-month data?

**Solution**
1、Query the yield rates of financial products in various projects in the Polkadot ecosystem based on keyword conditions.
2、Analyze the data of any address in the Polkadot ecosystem including wallet data analysis 、 data dashboard and so on.
3、The abnormal situation of large whale transactions in the Polkadot ecosystem recently
4、Ranking of important financings in the Polkadot ecosystem.
5、NFT and peripheral merchandise mall
6、Customized airdrop solution for many polkadot projects in future.
7、AI powered information service.
8、Provide equivalent data for on-chain transaction volume and transaction amount on a weekly, monthly, quarterly, semi-annual, and annual basis, as well as year-over-year and month-over-month growth data.

**Comment by semuelle** [comment]
Since you asked for private discussion, I will lock the conversation here for the moment. Let's continue this on Matrix.

**Review by semuelle — DISMISSED** [review]

**Review by takahser — DISMISSED** [review]
@lechangxu sorry for the delay here and thanks for answering the questions. Happy to approve as well.

File: applications/queryWeb3.md [hunk 10]
```diff
@@ -0,0 +1,292 @@
...
+## Development Status :open_book:
+
+Under construction
+
+### Overview
+
+- **Total Estimated Duration:** 4.5 months
+- **Full-Time Equivalent (FTE):**  60
+- **Total Costs:** 30,000 USD
+
```
Comment (keeganquigley) [inline applications/queryWeb3.md:193]: **DOT payment:** 50% of total costs.

**Review by keeganquigley — CHANGES_REQUESTED** [review]
Hi @lechangxu hopefully you're okay with accepting 50% of payment in vested DOT? If so feel free to accept my suggestion.

Additionally, we require all grantees to do KYC. Can you please use this [verification link](https://in.sumsub.com/idensic/l/#/uni_mjcB05gJZfh8dbr7) to complete it? Thanks and let me know if you run into any issues.

File: applications/queryWeb3.md (hunk 10 above)
Comment (keeganquigley) [inline applications/queryWeb3.md:193]:
```suggestion
**DOT payment:** 50% of total costs.
```

**Review by semuelle — APPROVED** [review]

**Review by keeganquigley — APPROVED** [review]
Thanks for the changes, I'll approve now and we will merge it once KYC has passed.

**Review by takahser — APPROVED** [review]
@lechangxu thanks for your answers. Happy to approve as well.

**Comment by keeganquigley** [comment]
Hi @lechangxu I wanted to also reach out here regarding https://github.com/w3f/Grants-Program/pull/2531 comments. Can you confirm this and give us an update on the project? Do you still have engineers to work on it? If we don't hear from you in a few weeks we will put up a vote to cancel it due to inactivity. Please let us know if you are still working on it. Thanks.

**Comment by lechangxu** [comment]
Yes,  we are currently building the project. please can you see? Here's the link:  https://github.com/orgs/Query-Web3/repositories

**Comment by lechangxu** [comment]
Please also check out https://queryweb3.com/! We are working hard on its development and kindly ask for your patience. Thank you all!

**Comment by keeganquigley** [comment]
Hi @lechangxu as seen in the above PR, M1 has been accepted. However, the foundation unfortunately wishes to cancel the remainder of the grant. This is due to the multitude of [issues](https://github.com/w3f/Grant-Milestone-Delivery/blob/master/evaluations/queryweb3_1_keeganquigley.md#general-notes) with the delivery. Therefore W3F is terminating M2 + M3 grant. I'm sorry to be the bearer of this bad news, but I'm glad we can at least pay you M1 in good faith, and let me know if you have any questions.

**Comment by lechangxu** [comment]
Hi @keeganquigley , 

Thank you for the update. While we are disappointed to hear that the grant for M2 and M3 will be terminated, we truly appreciate the Foundation’s decision to accept and pay for M1 in good faith.

Our team remains committed to the Polkadot ecosystem, and we look forward to any future opportunities where we can contribute again.

Regarding the M1 payment, could you please help me confirm if you have already received our invoice and if the format meets W3F’s specific requirements? Thank you for your support and guidance throughout this process.

**Pull request merged**